### ISSUE TITLE
Wrong indexes in specific cases

### ISSUE THREAD

**Issue by nneto** [kasramvd/SuffixTree#1]
At first I thought the problem only occurred when findall was set to True, until I got a wrong result when set to Falso too.

- In "test.py", setting s = "GATACATACA$" and findall = True:
Searching for "ATA" or "TAA" or "ACG", instead of receiving, respectively, [1, 5], [-1], and [-1] I'm getting [7, 3] for all of them;

- When findall = False:
"ACG" returns 1 instead of -1.

Maybe you are interested in knowing this.

**Comment by kasravnd** [comment]
@NNeto  Yeah, that's right! I'll check the issue as soon as possible. Thank you very much for your attention.
